index.html

```diff
@@ -394,11 +396,23 @@ <h4>Computation steps</h4>
                     <li>For <code>:after</code> pseudo elements, <a class="termref">User agents</a> MUST append <abbr title="Cascading Style Sheets">CSS</abbr>  textual content, without a space, to the textual content of the <code>current node</code>. </li>
                   </ul>
                 </li>
-                <li id="step2F.iii">For each child node of the <code>current node</code>:
+                <li id="step2F.iii">
+                  Determine the <code>rendered child nodes</code> of the <code>current node</code>:
                   <ol>
-                    <li id="step2F.iii.a">Set the <code>current node</code> to the child node.</li>
-                    <li id="step2F.iii.b">Compute the text alternative of the  <code>current node</code> beginning with step 2.  Set the <code>result</code> to that text alternative.</li>
-                    <li id="step2F.iii.c">Append the <code>result</code> to the <code>accumulated text</code>. </li>
+                    <li id="step2F.iii.a">If the <code>current node</code> has an attached [=shadow root=], set the <code>rendered child nodes</code> to be the child nodes of the [=shadow root=].</li>
+                    <li id="step2F.iii.b">Otherwise, if the <code>current node</code> is a [=slot=] with [=slot/assigned nodes=], set the <code>rendered child nodes</code> to be the [=slot/assigned nodes=] of the <code>current node</code>.</li>
+                    <div><details>
+                      <summary>Comment:</summary>
+                      <p>A [=slot=] element represents its [=slot/assigned nodes=], if any, and its contents (child nodes) otherwise.</p>
```

Comment:
I think this comment confuses me a bit?

In the step you wrote: if the `current node` is a slot, then set the `rendered` nodes to it's `assigned nodes`, if any.

But in the note, it sounds like the slot might have `child nodes` separate from it's `assigned nodes`.  So in this algorithm, if there are no `assigned nodes` but there are `child nodes` of a slot -- the `child nodes` are not added to the `rendered child node` list.

Maybe the note is unnecessary and the step needs to include what to do with slots without assigned nodes?

Let me know if I'm misunderstanding something, I'm new to thinking about web components.

Reply:
@jcsteh is correct; that's exactly what I was trying to communicate. [Here is a small CodePen](https://codepen.io/nolanlawson-the-selector/pen/rNvEZMX) to demonstrate.

Note that having _any_ nodes assigned to the `<slot>` (including an empty space) results in the assigned nodes being rendered. Otherwise, the child nodes (aka the default content) are rendered.

Perhaps there is a better way I can phrase it in the comment?

Reply:
I see -- tbh now I'm just not so sure a note is needed at all?

Reply:
Makes sense to me! I removed the comment.